Modules/System/Environment Information/src/EnvironmentInformationImpl.Codeunit.al

```diff
@@ -62,12 +68,27 @@ codeunit 3702 "Environment Information Impl."
         TestabilitySoftwareAsAService := EnableSoftwareAsAServiceForTest;
     end;
 
+    procedure SetTestabilityAppId(AppId: Text)
+    begin
+        TestabilityAppId := AppId;
+    end;
+
+    procedure SetTestabilityEnvironmentName(Name: Text)
+    begin
+        TestabilityEnvironmentName := Name;
+    end;
+
+    procedure SetTestMode(Mode: Boolean)
```

Comment:
It would be better to use an internal event so we can bind to it from test. It is cleaner because we can bind/unbind the manual subscriber from the test. With the global the setting will last for the entire session and it can affect the other tests. We cannot cleanup in case of the failure, thus a failing test could have set the value and it would stay for all other tests that will run within that session.

Reply:
With this approach we would need a new codeunit in which you can set the parameters and bind it.

Reply:
FYI: remember that internals should not be exposed directly to test code. It should be exposed to a test library which then has a public method for the tests to use.